### PR TITLE
Lms 1650   create error page filters

### PR DESCRIPTION
**Pull request by stephen-oh**
**Description**
- Create error page filters

**PR Type**
- Enhancement

### CODE REVIEW

File: classes/form/reportfilter_form.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once($CFG->libdir.'/formslib.php');
+
+/**
+ * The contact form to the site's Data Protection Officer
+ *
+ * @copyright 2020 onwards Stephen O'Hara
```
Comment (lzabo) [inline classes/form/reportfilter_form.php]: * @author     Stephen O'Hara \<stephen.ohara@learningpool.com>
 * @copyright  2020 Learning Pool Ltd \<http://learningpool.com>

File: report.php [hunk 2]
```diff
@@ -49,47 +49,48 @@
...
-        $where .= ' AND x.response = :response';
+        $where[] = 'x.response = :response';
         $params['response'] = $fromform->response;
     }
 
     if (!empty($fromform->datefrom)) {
         $datefrom = make_timestamp($fromform->datefrom['year'], $fromform->datefrom['month'], $fromform->datefrom['day']);
-        $where .= ' AND x.timecreated >= :datefrom';
+        $where[] = 'x.timecreated >= :datefrom';
         $params['datefrom'] = $datefrom;
```
Comment (lzabo) [inline report.php:73]: should be $fromform->datefrom ?

File: report.php [hunk 3]
```diff
@@ -49,47 +49,48 @@
...
     }
 
     if (!empty($fromform->dateto)) {
         $dateto = make_timestamp($fromform->dateto['year'], $fromform->dateto['month'], $fromform->dateto['day']);
-        $where .= ' AND x.timecreated <= :dateto';
+        $where[] = 'x.timecreated <= :dateto';
         $params['dateto'] = $dateto;
```
Comment (lzabo) [inline report.php:79]: should be $fromform->dateto?

File: classes/form/reportfilter_form.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once($CFG->libdir.'/formslib.php');
+
+/**
+ * The contact form to the site's Data Protection Officer
```
Comment (gordonmacqueen-lp) [inline classes/form/reportfilter_form.php]: contact form?

File: report.php [hunk 5]
```diff
@@ -49,43 +49,48 @@
...
+$sql = "SELECT x.id, x.errortype AS type, x.eventname, u.firstname, u.lastname, x.contextid, x.response, x.timecreated
+          FROM {logstore_xapi_failed_log} x
+     LEFT JOIN {user} u
+            ON u.id = x.userid
+         $where";
 $results = $DB->get_records_sql($sql, $params, $page*$perpage, $perpage);
-$count = $DB->count_records('logstore_xapi_failed_log');
+
+$sql = "SELECT COUNT(id)
+          FROM {logstore_xapi_failed_log} x
```
[the remark this answers is not in the document]
Comment (stephen-oh) [reply report.php:92]: This would require 2 where clauses, I kept this in for simplicity

**Review by gordonmacqueen-lp — APPROVED** [review]

**Pull request merged**